tests/integration/acp.go

```diff
@@ -161,13 +159,13 @@ type AddDocActorRelationship struct {
 	// The target public identity, i.e. the identity of the actor to tie the document's relation with.
 	//
 	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
-	TargetIdentity int
+	TargetIdentity identRef
```

Comment:
question: why do these need to be custom types now, instead of just int indexes? Doesn't `optional` struct work if you keep them both as separate identities instead of making them 1D?

Reply:
with just an index you can not differenciate between a node identity and a user identity (e.g. an identity provided externally).

Reply:
Answered here: https://github.com/sourcenetwork/defradb/pull/3125#discussion_r1814847264